### DISCUSSION TITLE
Run Build Not Working in Flynt 2.0.1

### DISCUSSION THREAD

**Discussion by purchasingtoolpak** [flyntwp/flynt#527] [Q&A] [answered]
When I run NPM run build - no files are added the dist folder - it remains empty. 

When I run NPM run start - dist folder shows a subfolder named "hot" - in the folder is the domain as https://localhost:5173. I am using Local for WordPress on my desktop.

The stylesheet files in the WordPress header show up like this:

```
<link rel='stylesheet' id='Flynt/assets/main-css' href='http://localhost:5173/assets/main.scss' media='all' />
--
<link rel='stylesheet' id='Flynt/assets/print-css' href='http://localhost:5173/assets/print.scss' media='print' />
```

When you click on the main.scss file it shows the files that should be compiled:
```
@import
  'styles/reset',
  'styles/variables',
  'styles/fonts',
  'styles/base',
  'styles/columns';

// stylelint-disable-next-line scss/at-import-no-partial-leading-underscore, scss/at-import-partial-extension-blacklist
@import '../Components/**/_style.scss';
```
The vite.config file seems to have the proper entries to build the "dist" folder and files.

The issue is that I am not able to deploy the website using wp-vivid. I never had this issue in Flynt 1.0.

Any debugging help would be appreciated.

Regards,

Dave

**Comment by purchasingtoolpak** [comment]
I created a second website using Flynt 2.01. It has the same issues:
1. Run Build does not compile sass files to main.css in dist folder.
2. Stylesheet does not have correct domain name in the WordPress header. Domain shown as localhost:1234 versus http://domain-name.local.

Thanks in advance for any assistance.

Regards,

Dave

**Reply by timohubois** [reply]
@purchasingtoolpak 

> 1. Run Build does not compile sass files to main.css in dist folder.

Please take a look on the 1st section of https://github.com/flyntwp/flynt/discussions/527#discussioncomment-8027231.

> 2.  Stylesheet does not have correct domain name in the WordPress header. Domain shown as localhost:1234 versus http://domain-name.local/.

If the `dist` folder contains a file with the name `hot` the domain should be `localhost` and if not it should be the domain of the WordPress instance. Please also take a look at the 2nd section of https://github.com/flyntwp/flynt/discussions/527#discussioncomment-8027231.

**Comment by timohubois** [comment]
Hey @purchasingtoolpak,

do you use Windows, macOS or Linux?

We have some other open issues in relation to Windows and I planned in to dive deeper into them today.

**Reply by purchasingtoolpak** [reply]
Windows 11.

**Comment by timohubois** [comment]
@purchasingtoolpak

> When I run NPM run build - no files are added the dist folder - it remains empty.

I found some time and installed [Local](https://localwp.com) and [Node.js](https://nodejs.org/en/download) on a fresh machine running Windows 11. After that I opened the command prompt (CMD), navigated to the theme folder and did the [install](https://github.com/flyntwp/flynt#install).

Then I ran npm start (inside the theme folder). At first it didn't work because Node.js was running in [production](https://nodejs.org/en/learn/getting-started/nodejs-the-difference-between-development-and-production) and devDependencies were not installed during the initial installation.

After setting Node.js to development by running `set NODE_ENV=development` inside the command prompt (CMD), I had to do an `npm i` again to complete the installation including the required devDependencies.

`npm start` worked now 💪 and then I tested [HMR](https://vitejs.dev/guide/features.html#hot-module-replacement) – it didn't work out-of-the-box. But I was able to fix this by running `npm start` in a new command prompt (CMD) started as **administrator**.

The next step was to test `npm run build`, which worked fine and the `dist` folder contains the required files.

<hr>

> When I run NPM run start - dist folder shows a subfolder named "hot" - in the folder is the domain as https://localhost:5173. I am using Local for WordPress on my desktop.

The hot file is a placeholder and needed for [HMR](https://vitejs.dev/guide/features.html#hot-module-replacement). When it exists styles and scripts are loaded in a different way and they **don’t** get the domain from the const `wordpressHost` inside the `vite.config.js` file.

<hr>

> The issue is that I am not able to deploy the website using wp-vivid. I never had this issue in Flynt 1.0.

Unfortunately I have no experience with wp-vivid and I don’t understand what exactly does not work with wp-vivid here and if this is related to flynt itself?

**Reply by purchasingtoolpak** [reply]
wp-vivid is not important - I was trying to say without the build css files I can't deploy the website.  Which you already understood.

**Reply by purchasingtoolpak** [reply]
Thanks for the quick response.
I ran CMD as administrator and changed to NODE_ENV to development, then ran npm i - unfortunately on 2 different sites they did not build the required CSS files. I'm just get the "hot" folder. 

Maybe I missed something, but nor working here.

**Reply by timohubois** [reply]
@purchasingtoolpak do you use the same setup as I did? And have you tested it exactly the way as described before?

**Reply by purchasingtoolpak** [reply]
Setup: Windows 11, Local, Node.js, Flynt 2.0.1, CMD terminal as Admin, template directory
Testing: set NODE_ENV=development, npm i, npm start, npm run build 

Npm start works but runs off of the hot folder. The flynt page stylesheet header looks like this:
```
<link rel='stylesheet' id='Flynt/assets/main-css' href='http://atliberty.local/wp-content/themes/flynt2/assets/main.scss' media='all' />
<link rel='stylesheet' id='Flynt/assets/print-css' href='http://atliberty.local/wp-content/themes/flynt2/assets/print.scss' media='print' />
```
Not sure what I am missing.

**Reply by timohubois** [reply]
@purchasingtoolpak thanks for your update.

> Npm start works but runs off of the hot folder. 

Does this mean that while you do a `npm start` the `dist`  folder keeps empty and **does not** get a `hot` file inside it?

<hr>

>The flynt page stylesheet header looks like this:
> 
> ```
> <link rel='stylesheet' id='Flynt/assets/main-css' href='http://atliberty.local/wp-content/themes/flynt2/assets/main.scss' media='all' />
> <link rel='stylesheet' id='Flynt/assets/print-css' href='http://atliberty.local/wp-content/themes/flynt2/assets/print.scss' media='print' />
> ```

This output is the case when no `hot` file nor a `mainfest` file can be found inside the `dist` folder ([Assets.php](https://github.com/flyntwp/flynt/blob/4ee06e3ddba4b09cb361c168536d87ae7934e726/lib/Utils/Asset.php#L99C14-L99C14)).

<hr>

In total it sounds like a really strange behavior and I can’t reproduce it at the moment. Have you tried to do a [build:production](https://github.com/flyntwp/flynt/blob/4ee06e3ddba4b09cb361c168536d87ae7934e726/package.json#L15C15-L15C15) by running `npm run build:production` only? Will this create files inside the `dist` folder? If not, can you post what the output of the cmd is showing after executing this command?

**Comment by purchasingtoolpak** [comment]
Thanks for that tip - run build:production - all the files are generated in the dist. folder. However, as soon as I run "npm run start" - the folder is emptied and only shows the "hot" folder. 

I tried to deploy the site and had issues with the stylesheet header. I am going to try again on Monday.

**Reply by timohubois** [reply]
@purchasingtoolpak great to read, that `run build:production` builds the files inside the `dist` folder.

 `run build:production` is the last step when executing `run build`, because this will execute npm-scripts sequentially (firstly `checkVersion` then `lint` and latest `build:production` see [package.json](https://github.com/flyntwp/flynt/blob/4ee06e3ddba4b09cb361c168536d87ae7934e726/package.json#L14C1-L15C1). 

In relation to this, the problem that the `dist` folder keeps empty is maybe related to `checkVersion`, `lint` or to the used `run-s` command. 

As written before on my fresh installation the problems are not reproducible https://github.com/flyntwp/flynt/discussions/527#discussioncomment-8027231 and it must be a specific problem on your environment. 

Now I’m asking me, what is the output in the cmd when you do a `run build`?

You also try to execute each of the `build` commands from the [package.json](https://github.com/flyntwp/flynt/blob/4ee06e3ddba4b09cb361c168536d87ae7934e726/package.json#L14C1-L15C1) manually, to figure out which may causes an issue in your specific case.

<hr>

> However, as soon as I run "npm run start" - the folder is emptied and only shows the "hot" folder.

This is the correct behavior. `npm run start` should only be used while development. For production executing `run build` once is enough in the most cases.

**Reply by purchasingtoolpak** [reply]
OK, I was able to deploy the site and it worked well except for the Hero Slider which I will address. 

With npm run build : production I can develop a Flynt 2.0 site and deploy. Still don't understand why npm run build does not work. If there are any other tests you think I should run let me know.

Appreciate your assistance.

**Reply by purchasingtoolpak** [reply]
I read your above note in more detail. Here is the terminal output after "run build" command.
> flynt@2.0.1 build
> run-s --silent checkVersion lint build:production

node: 20.10.0
standard: Use JavaScript Standard Style (https://standardjs.com)
standard: Run `standard --fix` to automatically fix some problems.
  C:\Users\djmak\Projects\Starter Site\WordPress\app\public\wp-content\themes\flynt2\Components\HeroSlider\script.js:11:1: Expected indentation of 4 spaces but found 2. (indent)
  C:\Users\djmak\Projects\Starter Site\WordPress\app\public\wp-content\themes\flynt2\Components\HeroSlider\script.js:11:31: Arrow function should not return assignment. (no-return-assign)
PS C:\Users\djmak\Projects\Starter Site\WordPress\app\public\wp-content\themes\flynt2>

I will also run the build manually as you suggest.

**Reply by timohubois** [reply]
Hi @purchasingtoolpak the issue is related to the linting in combination with the Premium Component “HeroSlider” and its current release – an update of the PremiumComponents will be published soon (it includes also some other minor changes of other components).

Technically the component should work as aspected and in this case it's just the syntax of the `script.js `file of the component “HeroSlider”.

> C:\Users\djmak\Projects\Starter Site\WordPress\app\public\wp-content\themes\flynt2\Components\HeroSlider\script.js:11:1: Expected indentation of 4 spaces but found 2. (indent)

Please make sure that line 11 has an indentation of 4 spaces instead of 2. Executing `npm run lintFix:scripts` should do this. If not please fix it manually.

<hr>

> C:\Users\djmak\Projects\Starter Site\WordPress\app\public\wp-content\themes\flynt2\Components\HeroSlider\script.js:11:31: Arrow function should not return assignment. (no-return-assign)

Replace in line 11: `initSlider(refs, data).then(x => swiper = x)` with `initSlider(refs, data).then((x) => { swiper = x })`.

<hr>

If you do not want use linting you can also remove lint from the build script `run-s --silent checkVersion lint build:production` > `run-s --silent checkVersion build:production`.

Let me know if this works for you?

**Reply by timohubois** [reply]
@purchasingtoolpak did you got it working?

**Reply by purchasingtoolpak** [reply, marked as answer]
I upgraded to 2.1 and am having no issues with run build using Windows 11.
